Dockerfile

```diff
@@ -19,156 +20,330 @@
+    && rm -rf /opt/scancode/thirdparty \
+    # Run scancode once to generate indexes as superuser
+    && /opt/scancode/bin/scancode --version
 
-COPY . /usr/local/src/ort
+#------------------------------------------------------------------------
+# ORT
+FROM nodebuild as ortbuild
+
+# Set this to the version ORT should report.
+ARG ORT_VERSION="DOCKER-SNAPSHOT"
 
+COPY . /usr/local/src/ort
 WORKDIR /usr/local/src/ort
 
-# Gradle build.
-ARG ORT_VERSION
+# Prepare Gradle
 RUN --mount=type=cache,target=/tmp/.gradle/ \
-    GRADLE_USER_HOME=/tmp/.gradle/ && \
-    scripts/import_proxy_certs.sh && \
-    scripts/set_gradle_proxy.sh && \
-    sed -i -r 's,(^distributionUrl=)(.+)-all\.zip$,\1\2-bin.zip,' gradle/wrapper/gradle-wrapper.properties && \
-    sed -i -r '/distributionSha256Sum=[0-9a-f]{64}/d' gradle/wrapper/gradle-wrapper.properties && \
-    ./gradlew --no-daemon --stacktrace -Pversion=$ORT_VERSION :cli:distTar :helper-cli:startScripts
-
-FROM adoptopenjdk:11-jdk-hotspot-focal AS run
-
-ENV \
-    # Package manager versions.
-    BOWER_VERSION=1.8.8 \
-    CARGO_VERSION=0.54.0-0ubuntu1~20.04.1 \
-    COCOAPODS_VERSION=1.11.2 \
-    COMPOSER_VERSION=1.10.1-1 \
-    CONAN_VERSION=1.43.2 \
-    GO_DEP_VERSION=0.5.4 \
-    GO_VERSION=1.16.5 \
-    HASKELL_STACK_VERSION=2.1.3 \
-    NPM_VERSION=7.20.6 \
-    PYTHON_PIPENV_VERSION=2018.11.26 \
-    PYTHON_VIRTUALENV_VERSION=15.1.0 \
-    SBT_VERSION=1.6.1 \
-    YARN_VERSION=1.22.10 \
-    # SDK versions.
-    ANDROID_SDK_VERSION=6858069 \
-    # Installation directories.
-    ANDROID_HOME=/opt/android-sdk \
-    GOPATH=$HOME/go
-
-ENV DEBIAN_FRONTEND=noninteractive \
-    PATH="$PATH:$HOME/.local/bin:$GOPATH/bin:/opt/go/bin:$GEM_PATH/bin:/opt/ort/bin"
+    . /opt/nodejs/nvm.sh \
+    && GRADLE_USER_HOME=/tmp/.gradle/ \
+    && scripts/import_proxy_certs.sh \
+    && scripts/set_gradle_proxy.sh \
+    && sed -i -r 's,(^distributionUrl=)(.+)-all\.zip$,\1\2-bin.zip,' gradle/wrapper/gradle-wrapper.properties \
+    && sed -i -r '/distributionSha256Sum=[0-9a-f]{64}/d' gradle/wrapper/gradle-wrapper.properties \
+    && ./gradlew --no-daemon --stacktrace -Pversion=$ORT_VERSION :cli:distTar :helper-cli:startScripts
+
+RUN mkdir -p /opt/ort \
+    && tar xf /usr/local/src/ort/cli/build/distributions/ort-$ORT_VERSION.tar -C /opt/ort --strip-components 1 \
+    && cp -a /usr/local/src/ort/scripts/*.sh /opt/ort/bin/ \
+    && cp -a /usr/local/src/ort/helper-cli/build/scripts/orth /opt/ort/bin/ \
+    && cp -a /usr/local/src/ort/helper-cli/build/libs/helper-cli-*.jar /opt/ort/lib/ \
+    && cd \
+    && rm -rf /usr/local/src
+
+#------------------------------------------------------------------------
+# Main container
+FROM eclipse-temurin:11-jdk-focal
+
+ENV LANG=en_US.UTF-8
+ENV LANGUAGE=en_US:en
+ENV LC_ALL=en_US.UTF-8
+ENV HOME=/workspace
+
+# Run with non privileged user
+RUN groupadd --gid 1000 ort \
+    && useradd --uid 1000 --gid ort --shell /bin/bash --home-dir /workspace --create-home ort
```

Comment:
Note to myself: Think about whether this eventually causes some problems when running the images with a custom user.

Reply:
Changed the way user is created. Now can be customized on build stage.